cfme/physical/provider/__init__.py

```diff
@@ -31,18 +40,45 @@ def __init__(
 
     @variable(alias='db')
     def num_server(self):
-        pass
+        provider = self.appliance.rest_api.collections.providers.find_by(name=self.name)[0]
+        num_server = 0
+        for server in self.appliance.rest_api.collections.physical_servers:
+            if server['ems_id'] == provider.id:
+                num_server += 1
+        return num_server
 
     @num_server.variant('ui')
     def num_server_ui(self):
-        pass
+        try:
+            num = self.get_detail("Relationships", 'Physical Servers')
+        except sel.NoSuchElementException:
+            logger.error("Couldn't find number of hosts using key [Hosts] trying Nodes")
+            num = self.get_detail("Relationships", 'Nodes')
+        return int(num)
+
+    def delete(self, cancel=True):
+        """
+        Deletes a provider from CFME
+
+        Args:
+            cancel: Whether to cancel the deletion, defaults to True
+        """
+        view = navigate_to(self, 'Details')
+        item_title = version.pick({'5.9': 'Remove this Infrastructure Provider from Inventory',
+                                   version.LOWEST: 'Remove this Infrastructure Provider'})
+        view.toolbar.configuration.item_select(item_title.format(self.string_name),
+                                               handle_alert=not cancel)
+        if not cancel:
+            msg = ('Delete initiated for 1 {} Provider from '
+                   'the {} Database'.format(self.string_name, self.appliance.product_name))
+            view.flash.assert_success_message(msg)
 
```

Comment:
I'd like to see an `else` clause here asserting no flash error messages and that the expected view is still displayed